### ISSUE TITLE
Customized block template with links, actions and configurations

### ISSUE THREAD

**Issue by unruledboy** [alyssaxuu/flowy#23]
Ok, this maybe an overhaul feature that consists of:

- developer can define any number of blocks (currently it is doing similar thing in `function snapping(drag, first)` ), may provide a function to do so
- each block can have its own template
- each template can have any number of links / actions (like the current "new visitor")
- each link/action can trigger different user-defined function
- and the right hand property window can also be customized per block, with good support of dynamic lists etc.

**Comment by alyssaxuu** [comment]
Right, so that would be using the current demo layout / style, and just allowing the developer to automatically generate blocks that can be dragged w/o having to individually stylize them?

Not sure about this, the demo is simply showing the potential of the tool, and it's not really my objective for people to use it as a base for their WebApps, specially if the styles and layout and everything is the same (obviously no problem if people do so, but it's not my intention).

Regarding triggering a user-defined function and whatnot, that's why there's the `flowy.output()` method, which lets you see the different properties of the blocks you place and with that information you can make it run any number of functions.

The right hand property window was simply an example of how each block could be customized property-wise, although I didn't actually add in the functionality as it's more of a proof of concept than a production-ready product (and I didn't feel there was really a need for it as the blocks didn't actually do anything either).

Basically I think I could either make a starter project that people can build upon, or simply leave those things you mentioned up to the developer (since I already provide the tools through methods and callbacks to recreate the behaviour you've described).

**Comment by unruledboy** [comment]
@alyssaxuu it's good to have a dedicated engine, and on top of that, have an implementation, like the way you are doing with the demo.

The thing is, most of the developers who would like to use this engine, would end up having a WYSIWYG designer, even each may have some difference in some way, but a starter project will go a long way, without lots of repeated impl.

**Comment by alyssaxuu** [comment]
There are no plans to create a boilerplate for this library at this time, so I'm closing the issue.
